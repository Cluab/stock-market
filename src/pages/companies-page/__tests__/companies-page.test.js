import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { MemoryRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import CompaniesPage from '../Companies-page';
import '@testing-library/jest-dom/extend-expect';

const middlewares = [thunk];
const initialState = [[{ symbol: 'GOOG', revenue: 50 }], [{ symbol: 'MSFT', revenue: 75 }], [{ symbol: 'AAPL', revenue: 100 }]];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ stocks: initialState });
describe('CompaniesPage component', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <CompaniesPage />
          </Router>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the Header and CompaniesList components', () => {
    const { getByTestId } = render(
      <Provider store={store}><Router><CompaniesPage /></Router></Provider>,
    );
    expect(getByTestId('header')).toBeInTheDocument();
    expect(getByTestId('companies-list')).toBeInTheDocument();
  });
});
