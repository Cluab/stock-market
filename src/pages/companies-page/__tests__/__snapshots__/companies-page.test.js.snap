// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompaniesPage component should match with snapshot 1`] = `
Array [
  <header
    className="header"
    data-testid="header"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa-solid fa-gear"
      />
    </button>
    <h1>
      stock market
    </h1>
  </header>,
  <a
    href="/details/GOOG"
    onClick={[Function]}
  >
    <div
      className="main-comp-section"
      data-testid="companies-list"
    >
      <span
        className="fa-solid fa-money-bill-trend-up"
      />
      <div
        className="main-comp-name"
      >
        <h2>
          GOOG
        </h2>
        <span>
          50
          / Revenue
        </span>
      </div>
    </div>
  </a>,
  <div
    className="main-page-bar"
  >
    <p>
      US STOCK MARKET
    </p>
  </div>,
  <div
    className="comp-name-list"
  >
    <a
      href="/details/MSFT"
      onClick={[Function]}
    >
      <div>
        <span
          className="fa-solid arrow-stock fa-circle-arrow-right"
        />
        <i
          className="fa-solid fa-money-bill-trend-up"
        />
        <h3
          data-testid="test-symbol"
        >
          MSFT
        </h3>
        <span
          data-testid="test-revenue"
        >
          75
        </span>
      </div>
    </a>
    <a
      href="/details/AAPL"
      onClick={[Function]}
    >
      <div>
        <span
          className="fa-solid arrow-stock fa-circle-arrow-right"
        />
        <i
          className="fa-solid fa-money-bill-trend-up"
        />
        <h3
          data-testid="test-symbol"
        >
          AAPL
        </h3>
        <span
          data-testid="test-revenue"
        >
          100
        </span>
      </div>
    </a>
  </div>,
]
`;
